docs/orchestrating-elastic-stack-applications/beat.asciidoc

```diff
@@ -0,0 +1,403 @@
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: filebeat-quickstart
+spec:
+  type: filebeat
+  version: {version}
+  elasticsearchRef:
+    name: elasticsearch-quickstart
+  config:
+    filebeat.inputs:
+    - type: container
+      paths:
+      - /var/log/containers/*.log
+  daemonSet:
+    podTemplate:
+      spec:
+        dnsPolicy: ClusterFirstWithHostNet
+        hostNetwork: true
+        securityContext:
+          runAsUser: 0
+        containers:
+        - name: filebeat
+          volumeMounts:
+          - name: varlogcontainers
+            mountPath: /var/log/containers
+          - name: varlogpods
+            mountPath: /var/log/pods
+          - name: varlibdockercontainers
+            mountPath: /var/lib/docker/containers
+        volumes:
+        - name: varlogcontainers
+          hostPath:
+            path: /var/log/containers
+        - name: varlogpods
+          hostPath:
+            path: /var/log/pods
+        - name: varlibdockercontainers
+          hostPath:
+            path: /var/lib/docker/containers
+EOF
+----
+
+. Monitor Beats
++
+Retrieve details about the Filebeat:
++
+[source,sh]
+----
+kubectl get beat
+----
++
+[source,sh,subs="attributes"]
+----
+NAME                  HEALTH   AVAILABLE   EXPECTED   TYPE       VERSION   AGE
+filebeat-quickstart   green    3           3          filebeat   {version}     2m
+----
++
+List all the Pods belonging to given Beat:
++
+[source,sh]
+----
+kubectl get pods --selector='beat.k8s.elastic.co/name=filebeat-quickstart-beat-filebeat'
+----
++
+[source,sh]
+----
+NAME                                      READY   STATUS    RESTARTS   AGE
+filebeat-quickstart-beat-filebeat-tkz65   1/1     Running   0          3m45s
+filebeat-quickstart-beat-filebeat-kx5jt   1/1     Running   0          3m45s
+filebeat-quickstart-beat-filebeat-nb6qh   1/1     Running   0          3m45s
+----
++
+. Access logs for one of the Pods
++
+[source,sh]
+----
+kubectl logs -f filebeat-quickstart-beat-filebeat-tkz65
+----
+
+. Access logs ingested by Filebeat
++
+- follow the Elasticsearch deployment link:k8s-deploy-elasticsearch.html[guide] and run:
++
+[source,sh]
+----
+curl -u "elastic:$PASSWORD" -k "https://localhost:9200/filebeat-*/_search"
+----
++
+- or follow the Kibana deployment link:k8s-deploy-kibana.html[guide], log in and go to *Kibana* > *Discover*.
+
+. Inspect generated Filebeat configuration
++
+[source,sh]
+----
+kubectl get secret filebeat-quickstart-beat-filebeat-config -o jsonpath='{.data.beat\.yml}' | base64 --decode
+----
++
+[source,sh]
+----
+filebeat:
+  inputs:
+  - paths:
+    - /var/log/containers/*.log
+    type: container
+output:
+  elasticsearch:
+    hosts:
+    - https://elasticsearch-quickstartquickstart-es-http.default.svc:9200
+    password: <redacted>
+    ssl:
+      certificate_authorities:
+      - /mnt/elastic-internal/elasticsearch-certs/ca.crt
+    username: default-filebeat-quickstart-beat-user
+----
```

Comment:
Do we need that part? We don't show this sort of example for other CRDs.
I understand why we'd want it to understand and deal with config merging but since we don't setup a default config besides the es output it feels unnecessary.

Reply:
Since we know about issues with `key: null` I wanted to hint how to look at the final config that Beat uses. But maybe that should be moved to troubleshooting section? I'll remove from here for now.

Reply:
Just leaving removed.